### ISSUE TITLE
textinput doesn't render styling correctly

### ISSUE THREAD

**Issue by simonasGit** [charmbracelet/bubbles#517]
```go
m.input.SetValue(lipgloss.NewStyle().
Background(lipgloss.Color("#00ff00")).
Render("hah"))
```
![image](https://github.com/charmbracelet/bubbles/assets/52699584/c2ab70c1-129c-46aa-9b22-e53ebf9d92af)
Instead of using `textinput.Model.TextStyle` to style the whole text, one may want to only style certain words for example.

I think it might be in [styleText](https://github.com/charmbracelet/bubbles/blob/master/textinput/textinput.go#L653) because escape sequences are being applied to a lipgloss styled string.

 - OS [Arch]
 - Shell [zsh, fish]
 - Terminal Emulator [Alacritty]

**Comment by meowgorithm** [comment]
Thanks for the issue! So as a baseline, we should definitely strip ansi from placeholders. Then the question becomes: should we allow placeholders to be styled as well? I _think_ so.

**Comment by simonasGit** [comment]
Yeah I guess if PlaceholderStyle and TextStyle are deprecated, if one wants a style for the whole string - `style.Render(m.ti.View())` or `m.ti.Placeholder = style.Render(text)`
